cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -564,9 +565,11 @@ func (s *Supervisor) onOpampConnectionSettings(_ context.Context, settings *prot
 func (s *Supervisor) waitForOpAMPConnection() error {
```

Comment:
Maybe my example is incorrect [here](https://go.dev/play/p/JdJ3wZbWcQT), but wouldn't this always immediately return and close the channel? I guess it's unclear to me how the timeout would ever be triggered now?

Reply:
`s.connectedToOpAMPServer <- struct{}{}` only returns if something else reads from the channel right? Consider this example: https://go.dev/play/p/JZ-endId3du

Same as yours but the reader is slower than the timeout (i.e. the connection takes longer to come in)

Reply:
ah right, this works because the channel is unbuffered so the write is waiting. Okay makes sense now. I've been too deep into the buffered world ha. Thanks for the expansion :)